abdiff/core/build_ab_images.py

```diff
@@ -1 +1,67 @@
-"""abdiff.core.build_ab_images"""
+import logging
+import os
+import shutil
+
+import docker
+from pygit2 import clone_repository
+from pygit2.enums import ResetMode
+
+from abdiff.core.utils import update_or_create_job_json
+
+logger = logging.getLogger(__name__)
+
+
+def build_ab_images(
```

Comment:
Leaving this comment here at the function level, given other comments in the logic itself.

Would it be worth seeing if an image is already built for a particular SHA?  In a comment below, noted that an image name could be `super-job-abc123` (or `transmogrifier-super-job-abc123` if we include that), and we'd know this as a combination of the job name + SHA.

As such, wonder if we could look for a docker image _already built_ with this image name.  If so, then it seems like we could skip (re)building the image and just be confident it exists, but still include in the `job.json` and return as normal (as if it had been created).

Reply:
Great idea, added a function to do this!